src/test/java/com/google/sps/ProfileBuilderServletTest.java

```diff
@@ -0,0 +1,228 @@
+  }
+
+  @Test
+  public void doPostRedirectsUrl() throws IOException, ServletException {
+    setRequestAsMentor();
+    profileBuilderServlet.doPost(request, response);
+
+    verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
+   }
+
+  @Test
+  public void doPostHasMissingUserInformation() throws IOException, ServletException {
+    DatastoreService dataService = DatastoreServiceFactory.getDatastoreService();
+ 
+    String userType = null;
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String mentorName = null;
+    when(request.getParameter("profName-input")).thenReturn(mentorName);
+    String company = null;
+    when(request.getParameter("company-input")).thenReturn(company);
+    String specialty = null;
+    when(request.getParameter("specialty-input")).thenReturn(specialty);
+
+    profileBuilderServlet.doPost(request, response);
+    verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
+    Assert.assertEquals(0, dataService.prepare(new Query("User")).countEntities(withLimit(10)));
+   }
+  
+  @Test
+  public void addMentorProfileThenCheckSize() throws IOException, ServletException {
+    DatastoreService dataService = DatastoreServiceFactory.getDatastoreService();
+    setRequestAsMentor();
+    profileBuilderServlet.doPost(request, response);
+
+    Assert.assertEquals(1, dataService.prepare(new Query("User")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void addingMultipleProfiles() throws IOException, ServletException {
+    DatastoreService dataService = DatastoreServiceFactory.getDatastoreService();
+    setRequestAsMentee();
+    profileBuilderServlet.doPost(request, response);
+
+    setRequestAsMentor();
+    profileBuilderServlet.doPost(request, response);
+
+    Assert.assertEquals(2, dataService.prepare(new Query("User")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void addMenteeProfileThenCheckSize() throws IOException, ServletException {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    setRequestAsMentee();
+    profileBuilderServlet.doPost(request, response);
+
+    Assert.assertEquals(1, ds.prepare(new Query("User")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void checkMenteeInfoIsSame() throws IOException, ServletException {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    String userType = "Mentee";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String menteeName = "Sandy Cane";
+    when(request.getParameter("name-input")).thenReturn(menteeName);
+    String school = "Stanford University";
+    when(request.getParameter("school-input")).thenReturn(school);
+    String major = "Computer Science";
+    when(request.getParameter("major-input")).thenReturn(major);
+
+    profileBuilderServlet.doPost(request, response);
+    // Filter schoolFilter = new FilterPredicate("school", FilterOperator.EQUAL, school);
```

Comment:
Can you remove the commented out code on line 167?

Reply:
Done